src/main/java/tds/testpackageconverter/converter/impl/TestPackageConverterServiceImpl.java

```diff
@@ -130,10 +142,48 @@ private Testspecification unzipAndRead(final ZipFile zipFile, final ZipEntry ent
         try {
             System.out.println("Reading file in zip: " + entry.getName());
             InputStream inputStream = zipFile.getInputStream(entry);
+
             return legacyXmlMapper.readValue(inputStream, Testspecification.class);
         } catch (IOException e) {
             log.error("An exception occurred: {}", e);
             throw new RuntimeException(e);
         }
     }
+
+    private TestPackageDiff unzipAndReadDiff(final ZipFile zipFile, final ZipEntry entry) {
+        try {
+            System.out.println("Reading file in zip: " + entry.getName());
+            InputStream inputStream = zipFile.getInputStream(entry);
+
+            return testPackageMapper.readValue(inputStream, TestPackageDiff.class);
+        } catch (IOException e) {
+            log.error("An exception occurred: {}", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+
+    private boolean isTestSpecification(final ZipFile zipFile, final ZipEntry entry) {
+        try {
+            InputStream inputStream = zipFile.getInputStream(entry);
+            String fileText = IOUtils.toString(inputStream, StandardCharsets.UTF_8.name());
+            return fileText.contains("<testspecification");
```

Comment:
A bit iffy on this method of checking if the test package is a diff or legacy testspec. We need a way to "peek" at the file to see what it is before we try and process it. I guess one option would be validating against an XSD, but that might be a little much.

Reply:
Does the spec have a version element we could look for? Might be nice to have.